cmd/docker/docker.go

```diff
@@ -50,6 +48,11 @@ func newDockerCommand(dockerCli *command.DockerCli) *cobra.Command {
 	flags.StringVar(&opts.ConfigDir, "config", cliconfig.ConfigDir(), "Location of client config files")
 	opts.Common.InstallFlags(flags)
 
+	// flags must be the top-level command flags, not cmd.Flags()
+	opts.Common.SetDefaultOptions(flags)
+	dockerCli.Initialize(opts)
+	dockerCli.Ping()
```

Comment:
ping @mlaventure since it's not related specifically to the API versioning, but also experimental.